OnepayMVCTest/Controllers/TransactionController.cs

```diff
@@ -23,38 +32,57 @@ public ActionResult Create()
                     amount: p.Price,
                     additionalData: null,
                     expire: 10
-                    ));
+                ));
             TransactionCreateResponse response;
             try
             {
-                response = Transaction.Create(shoppingCart);
-            } catch (Transbank.Onepay.Exceptions.TransbankException e)
+                response = Transaction.Create(shoppingCart, channelType);
+            }
+            catch (Transbank.Onepay.Exceptions.TransbankException e)
             {
-                return RedirectToAction("Error", "Message", new { error = e.Message });
+                return Json("{\"error\":" + e.Message + "}");
```

Comment:
Concatenating Strings to create JSON objects is a bad idea. For example if the message has the `"` character you will generate invalid JSON. Please use whatever way JSON is usually generated in .NET

Reply:
...which seems to be creating a hash table with the error element and then converting the hashtable to json

Reply:
@goncafa This should be fixed before merging though

Reply:
Code updated